config/routes.rb

```diff
@@ -9,6 +9,8 @@
   resource :pricing, only: :show, controller: :pricing
   resource :role, only: :new
   resources :businesses, except: :destroy
+  resources :notifications, only: %i[index update], controller: :notifications
+  resources :read_notifications, path: "notifications/read", only: :index, controller: :read_notifications
```

Comment:
```suggestion
  resources :notifications, only: %i[index update]
  namespace :notifications do
    resources :read, only: :index
  end
```

This will route to a `Notifications::ReadController` and I think is a little cleaner. Less code, and more conventional, at least!

Reply:
I will test locally. I thought I’d tried that early on and got unresolvable errors, but it may have been my syntax.

Reply:
FYI, the benefit of this type of routing is explained here: http://jeromedalbert.com/how-dhh-organizes-his-rails-controllers/

Also, you can check the output of the routes file by visiting `http://localhost:3000/rails/info/routes` or running `rails routes`

Reply:
I'm making progress on getting this to work, I think. It feels very convoluted though - and definitely not less code. So I'm wondering if I'm doing it wrong.

Reply:
Interesting - I thought you should be able to move the `ReadNotificationsController` code into `Notifications::ReadController` and rename the `views/` folders. What additional code is required?

Reply:
I've added a commit with the basic changes (I did eventually get it to work.) 

The main thing that seemed convoluted to me was needing to make both 'read' and 'notifications' folder within views/notifications. And how I have two partials for notifications instead of one, even though they should be virtually identical. Should I/could I create a component for notification and just have each _notification.html.erb partial import that component? Is there a better way to do it that I'm not thinking about? I don't want someone in the future to have to update two templates every time a small tweak is made to the general layout of a notification.

Also I didn't originally catch that if I used a namespace, I'd need to create a subfolder within controllers called notifications to hold `Notifications::ReadController`. The docs on routes are... not the clearest. But I think I'm getting up to speed.

Reply:
Ah, we definitely don't want that much duplicated logic. Great catch.

I'm fine with duplicating the `index` views, especially since we have slightly different logic going on. However, for the notification partials, we can re-use `app/views/notifications/_notification.html.erb` by explicitly rendering the partial from `app/views/notifications/read/index.html.erb:4`.

```ruby
<%= render partial: "notifications/notification", collection: @read %>
```

I also think we can rename `@read` to `@notifications` since we know the context based on the controller and view names.

Reply:
OK perfect, this is really helpful.